lib/_http_outgoing.js

```diff
@@ -90,11 +90,12 @@ exports.OutgoingMessage = OutgoingMessage;
 
 OutgoingMessage.prototype.setTimeout = function(msecs, callback) {
 
-  if (callback) {
-    if (typeof callback !== 'function')
-      throw new TypeError('callback must be a function');
+  if (typeof callback === 'function') {
     this.on('timeout', callback);
   }
+  else {
+    throw new TypeError('callback must be a function');
```

Comment:
didn't mean to bury this but realized that there was a bug here.

Want to continue the conversation at --> https://github.com/nodejs/node/pull/3539/files#r43200588